concordium-consensus/src/Concordium/GlobalState/BlockState.hs

```diff
@@ -1665,6 +1665,34 @@ class (BlockStateQuery m, PLTQuery (UpdatableBlockState m) m) => BlockStateOpera
         -- | The index of the new token and the updated block state.
         m (TokenIndex, UpdatableBlockState m)
 
+    -- | Update the token module state. If the update is not succesful, this returns Nothing.
+    --
+    --  PRECONDITION: The token identified by 'TokenIndex' MUST exist.
+    bsoUpdateTokenAccountModuleState ::
+        (PVSupportsPLT (MPV m)) =>
+        UpdatableBlockState m ->
+        -- | The token index to update.
+        TokenIndex ->
+        -- | The account to update.
+        AccountIndex ->
+        -- | The list of updates to the module state.
+        [(TokenStateKey, GSAccount.TokenAccountStateValueDelta)] ->
+        m (Maybe (UpdatableBlockState m))
```

Comment:
I wasn't sure about how we want to handle errors in the interface. This implementation uses `Maybe` to signal success/error. Once we can emit events, we might want to change this to `Either` to propagate the error message, or emit the events at the place the error occurs.

Reply:
Removed `TASVCreate` and changed the return type to `m (UpdatableBlockState m)`.